### ISSUE TITLE
Can we use the matrix factorization model locally by downloading it to our local?

### ISSUE THREAD

**Issue by nileshm11** [lm-sys/RouteLLM#39]
I am using a matrix factorization approach in a RAG application and would like to download this MF model to my local system for use. Is this possible? Additionally, could you provide an idea of how it works internally?

**Comment by iojw** [comment]
Yes, it is possible to download this from HuggingFace! https://huggingface.co/routellm/mf_gpt4_augmented. However, note that you will still have to generate embeddings for OpenAI as input for the time being - @thwu1 is working on adding support for other embedding models in #17!

Please refer to the paper for specific details for how it works: https://arxiv.org/abs/2406.18665

**Comment by nileshm11** [comment]
Could you please provide more details or share the implementation code for integrating this into a local system? Specifically, I'm interested in how to set up and use the model, including any necessary configurations and dependencies.
